pytest_fixtures/component/repository.py

```diff
@@ -51,3 +51,139 @@ def repo_setup():
     lce = entities.LifecycleEnvironment(organization=org).create()
     details = {'org': org, 'product': product, 'repo': repo, 'lce': lce}
     yield details
+
+
+def _simplify_repos(request, repos):
+    """This is an helper function that transforms repos_collection related fixture parameters into
+    the list that can be passed to robottelo.host_helpers.RepositoryMixins.RepositoryCollection
+    class constructor arguments. Using which it can create multiple repos with distro.
+
+    E.g: the parameters list to repos_collection fixture is:
+    [{
+        'distro': DISTRO_RHEL7,
+        'SatelliteToolsRepository': {},
+        'YumRepository': [{'url': settings.repos.yum_0.url},{'url': settings.repos.yum_6.url}]
+    }]
+
+    Then it removes disto from the dict above and simplifies/translates remaining dictionary into:
+    [
+        {'SatelliteToolsRepository': {}},
+        {'YumRepository': {'url': settings.repos.yum_0.url}},
+        {'YumRepository': {'url': settings.repos.yum_6.url}}
+    ]
+    So that the fixtures loop over it to create multiple repositories.
+
+    :returns: The tuple of distro of repositories(if given) and simplified repos
```

Comment:
s/into the list/into a list/
s/Using this it can create multiple repos with distro./It can create multiple repos per distro./
s/Then it removes distro from the dict above and simplifies/translates the remaining dictionary into/The fixture removes `distro` from the dict above and translates the remaining dictionary into/
s/So that the fixtures loop over it to create multiple repositories./Then the fixtures loops over it to create multiple repositories./
s/The tuple of distros of repositories(if given) and simplified repos/The tuple of distros of repositories (if given) and simplified repos/